src/frontend/context/ManuellJournalførContext.tsx

```diff
@@ -0,0 +1,634 @@
+import React from 'react';
+
+import { AxiosError } from 'axios';
+import createUseContext from 'constate';
+import { useHistory, useParams } from 'react-router';
+
+import {
+    byggDataRessurs,
+    byggFeiletRessurs,
+    byggHenterRessurs,
+    byggTomRessurs,
+    IDokumentInfo,
+    Ressurs,
+    RessursStatus,
+    AvsenderMottakerIdType,
+    kjønnType,
+    AvsenderMottaker,
+    byggSuksessRessurs,
+    IJournalpost,
+} from '@navikt/familie-typer';
+
+import { IOpprettBehandlingData, IOpprettEllerHentFagsakData } from '../api/fagsak';
+import {
+    BehandlingKategori,
+    Behandlingstype,
+    BehandlingUnderkategori,
+    BehandlingÅrsak,
+    IBehandling,
+} from '../typer/behandling';
+import { IFagsak } from '../typer/fagsak';
+import {
+    BrevkodeMap,
+    IDataForManuellJournalføring,
+    IRestJournalføring,
+    JournalpostKanal,
+} from '../typer/manuell-journalføring';
+import { Adressebeskyttelsegradering, IPersonInfo, PersonType } from '../typer/person';
+import { hentAktivBehandlingPåFagsak } from '../utils/fagsak';
+import familieDayjs from '../utils/familieDayjs';
+import { useApp } from './AppContext';
+
+const tomPerson = {
+    adressebeskyttelseGradering: Adressebeskyttelsegradering.UGRADERT,
+    familierelasjoner: [],
+    familierelasjonerMaskert: [],
+    fødselsdato: '',
+    kjønn: kjønnType.UKJENT,
+    navn: '',
+    personIdent: '',
+    type: PersonType.SØKER,
+};
+
+const tomAvsender = {
+    erLikBruker: false,
+    id: '',
+    land: '',
+    navn: '',
+    type: AvsenderMottakerIdType.UKJENT,
+};
+
+const erPersonTom = (person: IPersonInfo | undefined) => !person || !person.personIdent;
+
+const erAvsenderTom = (avsender: AvsenderMottaker | undefined) => !avsender || !avsender.navn;
+
+const patchTomFelt = (ress: Ressurs<IDataForManuellJournalføring>) => {
+    // we use tom object for person and avsender if they are not present in data
+    // because we need to use the objects to index the validation errors (See validaterData() for details)
+    if (ress.status === RessursStatus.SUKSESS) {
+        if (!ress.data.person) {
+            ress.data.person = tomPerson;
+        }
+        if (!ress.data.journalpost.avsenderMottaker) {
+            ress.data.journalpost.avsenderMottaker = tomAvsender;
+        }
+    }
+    return ress;
+};
```

Comment:
Try to give these a more intuitive name as discussed :)

Reply:
Ok